dace/sdfg/work_depth_analysis/work_depth.py

```diff
@@ -304,6 +304,11 @@ def sdfg_work_depth(sdfg: SDFG, w_d_map: Dict[str, Tuple[sp.Expr, sp.Expr]], ana
     :param w_d_map: Dictionary which will save the result.
     :param analyze_tasklet: Function used to analyze tasklet nodes.
     :param symbols: A dictionary mapping local nested SDFG symbols to global symbols.
+    :param detailed_analysis: If True, detailed analysis gets used. For each branch, we keep track of its condition
```

Comment:
Due to the performance impact, I would consider making this parameter default to `False` in the call signature. Thoughts @acalotoiu ?

Reply:
Sounds good to me!